apps/services/auth/admin-api/src/app/v2/delegations/test/delegation-admin.spec.ts

```diff
@@ -0,0 +1,332 @@
+    })
+
+    it('DELETE /delegation-admin/:delegation should throw error since id does not exist', async () => {
+      // Arrange
+      await createDelegationAdmin(currentUser)
+
+      const invalidId = faker.datatype.uuid()
+      // Act
+      const res = await getRequestMethod(
+        server,
+        'DELETE',
+      )(`/delegation-admin/${invalidId}`)
+
+      // Assert
+      expect(res.status).toEqual(204)
+
+      // Assert db
+      const delegationModel = await app.get(getModelToken(Delegation))
+      const deletedDelegation = await delegationModel.findAll()
+
+      expect(deletedDelegation).not.toBeNull()
+    })
+  })
+
+  describe('POST /delegation-admin', () => {
+    const toNationalId = '0101302399'
+    const fromNationalId = '0101307789'
+
+    let zendeskServiceApiSpy: jest.SpyInstance
+    let nationalRegistryApiSpy: jest.SpyInstance
+
+    let delegationModel: typeof Delegation
+    let delegationDelegationTypeModel: typeof DelegationDelegationType
+
+    beforeEach(async () => {
+      delegationModel = await app.get(getModelToken(Delegation))
+      delegationDelegationTypeModel = await app.get(
+        getModelToken(DelegationDelegationType),
+      )
+
+      await factory.createDomain({
+        name: 'd1',
+        apiScopes: [
+          {
+            name: 's1',
+            supportedDelegationTypes: [
+              AuthDelegationType.Custom,
+              AuthDelegationType.GeneralMandate,
+            ],
+          },
+        ],
+      })
+
+      mockZendeskService(toNationalId, fromNationalId)
+      mockNationalRegistryService()
+    })
+
+    const mockNationalRegistryService = () => {
+      nationalRegistryApiSpy = jest
+        .spyOn(nationalRegistryApi, 'getIndividual')
+        .mockImplementation(async (id) => {
+          const user = createNationalRegistryUser({
+            nationalId: id,
+          })
+
+          return user ?? null
+        })
+    }
+
+    const mockZendeskService = (
+      toNationalId: string,
+      fromNationalId: string,
+    ) => {
+      zendeskServiceApiSpy = jest
+        .spyOn(zendeskService, 'getTicket')
+        .mockImplementation((ticketId: string) => {
+          return new Promise((resolve) =>
+            resolve({
+              id: ticketId,
+              tags: [DELEGATION_TAG],
+              status: TicketStatus.Solved,
+              custom_fields: [
+                {
+                  id: ZENDESK_CUSTOM_FIELDS.DelegationToReferenceId,
+                  value: toNationalId,
+                },
+                {
+                  id: ZENDESK_CUSTOM_FIELDS.DelegationFromReferenceId,
+                  value: fromNationalId,
+                },
+              ],
+            }),
+          )
+        })
+    }
+
+    afterEach(() => {
+      zendeskServiceApiSpy.mockRestore()
+      nationalRegistryApiSpy.mockRestore()
+    })
+
+    afterAll(() => {
+      jest.restoreAllMocks()
+      app.cleanUp()
+    })
```

Comment:
I don't think that this is necessary. You are already calling `app.cleanUp` `afterEach` and you are restoring the mocks `afterEach`